multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+  stat_summary(
+    fun.data = "mean_sdl",  fun.args = list(mult = 1), 
+    geom = "pointrange", color = "black"
+    )
+    
+# Combine with box plot to add median and quartiles
+# Change color by groups
+e + geom_violin(aes(fill = dose), trim = FALSE) + 
+  geom_boxplot(width = 0.2)+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800", "#FC4E07"))+
+  theme(legend.position = "none")
+```
+
+#### 3.3.2 创建具有多个组的小提琴图
+
+```{r}
+e + geom_violin(
+  aes(color = supp), trim = FALSE,
+  position = position_dodge(0.9) 
+  ) +
+  geom_boxplot(
+    aes(color = supp), width = 0.15,
+    position = position_dodge(0.9)
+    ) +
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+### 3.4 点图
+
+关键函数：
+
+geom_dotplot()：创建堆叠的点，每个点代表一个观测值。主要参数：
+
+- stackdir：点堆叠的方向。 “上”（默认），“下”，“中心”，“整个中心”（居中，但点对齐）。
+
+- stackratio：堆叠点的距离。默认值为1，其中点与点之间仅接触。对于较小的重叠点，请使用较小的值。
+
+- olor, fill：点边界颜色和区域填充。
+
+- dotsize：相对于binwidth的点的直径，默认为1。
+
+对于小提琴图，通常将摘要统计信息添加到点图。
+
+#### 3.4.1 创建基本点图
+
+```{r}
+# Violin plots with mean points +/- SD
+e + geom_dotplot(
+  binaxis = "y", stackdir = "center",
+  fill = "lightgray"
+  ) + 
+  stat_summary(
+    fun.data = "mean_sdl", fun.args = list(mult=1), 
+    geom = "pointrange", color = "red"
+    )
+# Combine with box plots
+e + geom_boxplot(width = 0.5) + 
+  geom_dotplot(
+    binaxis = "y", stackdir = "center",
+    fill = "white"
+    ) 
+# Dot plot + violin plot + stat summary
+e + geom_violin(trim = FALSE) +
+  geom_dotplot(
+    binaxis='y', stackdir='center',
+    color = "black", fill = "#999999"
+    ) +
+  stat_summary(
+    fun.data="mean_sdl",  fun.args = list(mult=1), 
+    geom = "pointrange", color = "#FC4E07", size = 0.4
+    )
+```
+
+
+#### 3.4.2 创建具有多个组的点图
+
+```{r}
+# Color dots by groups
+e + geom_boxplot(width = 0.5, size = 0.4) +
+  geom_dotplot(
+    aes(fill = supp), trim = FALSE,
+    binaxis='y', stackdir='center'
+  )+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800"))
+# Change the position : interval between dot plot of the same group
+e + geom_boxplot(
+  aes(color = supp), width = 0.5, size = 0.4,
+  position = position_dodge(0.8)
+  ) +
+  geom_dotplot(
+    aes(fill = supp, color = supp), trim = FALSE,
+    binaxis='y', stackdir='center', dotsize = 0.8,
+    position = position_dodge(0.8)
+  )+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800"))+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+### 3.5 带状图
+
+带状图也称为一维散点图。当样本量较小时，与箱形图相比，带状图更合适。
+
+关键函数：
+
+geom_jitter()。关键参数：
+
+color, fill, size, shape：更改点的颜色，填充，大小和形状
+
+#### 3.5.1 创建一个基本的带状图
+
+- 按组更改点的形状和颜色
+
+- 调整抖动程度：position_jitter(0.2)
+
+- 添加摘要统计信息
+
+```{r}
+e + geom_jitter(
+  aes(shape = dose, color = dose), 
+  position = position_jitter(0.2),
+  size = 1.2
+  ) +
+  stat_summary(
+    aes(color = dose),
+    fun.data="mean_sdl",  fun.args = list(mult=1), 
+    geom = "pointrange",  size = 0.4
+    )+
+  scale_color_manual(values =  c("#00AFBB", "#E7B800", "#FC4E07"))
+```
+
+#### 3.5.2 创建具有多个组的带状图。
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.